packages/Pagination/src/Pagination.js

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import PropTypes from "prop-types";
+// import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import ArrowItem from "./components/ArrowItem/ArrowItem";
+import CurrentPageItem from "./components/CurrentPageItem/CurrentPageItem";
+import ElipsisItem from "./components/EllipsisItem/ElipsisItem";
+import PageItem from "./components/PageItem/PageItem";
+import Divider from "./components/Divider";
+import * as sc from "./Pagination.styles";
+
+const propTypes = {
+  /** The number of current active page */
+  currentPage: PropTypes.number.isRequired,
+  /** The number of other pages that will be visible around the current/active page (not hidden by elipsis). */
+  pagesOnEachSide: PropTypes.number,
+  /** Callback to be executed when current page is changed. */
+  onClick: PropTypes.func.isRequired,
+  /** The number of total pages. */
+  totalPages: PropTypes.number.isRequired,
+};
+
+const defaultProps = {
+  pagesOnEachSide: 2,
+};
+
+function Pagination(props) {
+  const { currentPage, totalPages, pagesOnEachSide, onClick, ...moreProps } = props;
+
+  const isCurrentPage = pageNumber => currentPage === pageNumber;
+  const isFirstPage = pageNumber => pageNumber === 1;
+  const isLastPage = pageNumber => pageNumber === totalPages;
+
+  const isNotEllipsed = (pageNumber, left, right) =>
+    (pageNumber === 2 && left === 3) || (pageNumber === totalPages - 1 && right === totalPages - 2);
+
+  const isRenderableSibling = (pageNumber, left, right) => left <= pageNumber && pageNumber <= right;
+
+  const renderPageElements = () => {
+    const delta = pagesOnEachSide;
+    const left = currentPage - delta;
+    const right = currentPage + delta;
+
+    const items = [];
+
+    let wasPreviousItemEllipsized = false;
+
+    for (let pageNumber = 1; pageNumber <= totalPages; pageNumber++) {
+      if (isCurrentPage(pageNumber)) {
+        items.push([<CurrentPageItem key={pageNumber} pageNumber={pageNumber} />, <Divider />]);
```

Comment:
My thinking is that with the divider as a component it is easier to adjust the styling. Instead of having to add &::after css for all of the elements and have to adjust each one when we want to make changes, it is all in conveniently in one place. Additionally, it means we don't have to set up specific styling for the Arrow Icons when the arrow is left or right, instead it will be rendered in the appropriate place using the component. What do you think?

Reply:
Personally I'd prefer to simplify the number of DOM elements rendered, for example using `:after` when we can. But this is not a big deal if it's hard to do